File: LV1/추억 점수.js
function solution(name, yearning, photo) {
  let memoryScores = [];
  let yearningObj = {};
  for (let i = 0; i < name.length; i++) {
    yearningObj[name[i]] = yearning[i];
  }
  for (let i = 0; i < photo.length; i++) {
    let score = 0;
    for (let j = 0; j < photo[i].length; j++) {
      if (yearningObj[photo[i][j]] !== undefined) {
        score += yearningObj[photo[i][j]];
      }
    }
    memoryScores.push(score);
  }
  return memoryScores;
}

File: LV1/추억 점수.java
import java.util.*;

class Solution {

  public int[] solution(String[] name, int[] yearning, String[][] photo) {
    int[] answer = new int[photo.length];
    HashMap<String, Integer> map = new LinkedHashMap<>();
    for (int i = 0; i < name.length; i++) {
      map.put(name[i], yearning[i]);
    }

    for (int i = 0; i < photo.length; i++) {
      String[] people = photo[i];
      int score = 0;
      for (int j = 0; j < people.length; j++) {
        String person = people[j];
        if (map.containsKey(person)) {
          score += map.get(person);
        }
      }
      answer[i] = score;
    }

    return answer;
  }
}
